weaver/processes/wps_package.py

```diff
@@ -2069,7 +2069,8 @@ def make_location_input(self, input_type, input_definition):
                         f"Directory reference resolution method for input [{input_id}] "
                         f"from location [{input_location}] did not produce any staged file."
                     )
-                input_location = out_dir
+                input_directory_name = os.path.basename(os.path.normpath(input_definition.url))
+                input_location = os.path.join(out_dir, input_directory_name)
```

Comment:
There are a few tests (ex: https://github.com/crim-ca/weaver/blob/master/tests/functional/test_wps_package.py#L2240 and others below) that validate dirs submitted as input to obtain a listing. I think this change will still allow them to work, but you could add the relevant `assert` in those tests (check end of each test validating dir/value format in output) to check that the produced listing contains the last directory name.